addons/sale_project/models/project.py

```diff
@@ -58,15 +58,16 @@ def action_view_so(self):
         return action_window
 
     def action_create_invoice(self):
-        if not self.sale_order_id and self.sale_order_id.invoice_status != 'to invoice':
-            raise UserError(_("The selected Sales Order should contain something to invoice."))
+        if not self.has_any_so_to_invoice:
+            raise UserError(_("The are is nothing to invoice in this project."))
         action = self.env["ir.actions.actions"]._for_xml_id("sale.action_view_sale_advance_payment_inv")
-        sols = self.sale_order_id.ids
+        so_ids = self.sale_order_id.ids
         for task in self.task_ids:
-            if task.sale_order_id.id not in sols:
-                sols += [task.sale_order_id.id]
+            if task.sale_order_id.id not in so_ids:
+                so_ids += [task.sale_order_id.id]
```

Comment:
We could do this in one line 😇 
```suggestion
        so_ids = (self.sale_order_id | self.task_ids.sale_order_id).filtered(lambda so: so.invoice_status == 'to invoice').ids
```

(Seems there is a bug in the diff of the suggestion) 😅